File: DataBindingDemo/app/src/main/java/com/junkchen/databindingdemo/MainActivity.java
package com.junkchen.databindingdemo;

import androidx.lifecycle.LiveData;
import androidx.databinding.DataBindingUtil;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;

import com.junkchen.databindingdemo.databinding.ActivityDataBinding;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
//        setContentView(R.layout.activity_data);
        ActivityDataBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_data);

        User user = new User("Junk", "Chen");
        binding.setUser(user);
        binding.setPresenter(new Presenter());

    }
}

File: DataBindingDemo/doubanmovie/src/main/java/com/junkchen/doubanmovie/MainActivity.kt
package com.junkchen.doubanmovie

import android.annotation.SuppressLint
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.junkchen.doubanmovie.view.Movie2Fragment

/**
 * Created by Junk Chen on 2018/12/17.
 */
class MainActivity : AppCompatActivity() {

    @SuppressLint("CommitTransaction")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
//        DataBindingUtil.setContentView<com.junkchen.doubanmovie.databinding.ActivityMainBinding>(this, R.layout.activity_main)
        supportFragmentManager.beginTransaction()
//                .add(R.id.fl_movie, MovieFragment.newInstance())
                .add(R.id.fl_movie, Movie2Fragment.newInstance())
                .commitNow()
    }
}
